src/main/java/org/jenkinsci/plugins/junitrealtimetestreporter/PipelineRealtimeTestResultAction.java

```diff
@@ -90,18 +93,43 @@ protected TestResult parse() throws IOException, InterruptedException {
         if (ws != null && ws.isDirectory()) {
             LOGGER.log(Level.FINE, "parsing {0} in {1} on node {2} for {3}", new Object[] {glob, workspace, node, run});
 
-            FlowNode node = context.get(FlowNode.class);
-            List<FlowNode> enclosingBlocks = JUnitResultsStepExecution
-                    .getEnclosingStagesAndParallels(requireNonNull(node));
+            FlowNode node;
+            TaskListener listener;
+            // lots of boilerplate code to get access to the flow node and listener but without saving
+            // the StepContext via xstream in case of restarts, we default to using step context if it's available though
+            if (context != null) {
+                node = context.get(FlowNode.class);
+                listener = context.get(TaskListener.class);
+            } else {
+                if (this.run instanceof FlowExecutionOwner.Executable) {
+                    FlowExecutionOwner.Executable executable = (FlowExecutionOwner.Executable) this.run;
+                    FlowExecutionOwner flowOwner = executable.asFlowExecutionOwner();
+                    if (flowOwner != null) {
+                        node = flowOwner.get().getNode(id);
```

Comment:
null check IIRC

Reply:
.get() is non null, the node is checked for null on line 121

Reply:
Oh right, I meant to say that you could consider calling `getOrNull` so as to fall back to `TaskListener.NULL` in more edge conditions. Not sure it would be helpful though.

Reply:
Ah right, done in the last commit, and simplified a bit